benchmarks/layeredflow/layeredflow.cc

```diff
@@ -0,0 +1,461 @@
+      this->model_dependence.viscosity = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.density = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.compressibility = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.specific_heat = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = MaterialModel::NonlinearDependence::none;
+    }
+
+
+    template <int dim>
+    double
+    LayeredFlowMaterial<dim>::get_beta() const
+    {
+      return beta;
+    }
+
+    template <int dim>
+    double
+    LayeredFlowMaterial<dim>::get_epsilon() const
+    {
+      return epsilon;
+    }
+
+    template <int dim>
+    LayeredFlowBoundary<dim>::LayeredFlowBoundary ()
+      :
+      beta (0),
+      epsilon (0)
+    {}
+
+
+
+    template <>
+    Tensor<1,2>
+    LayeredFlowBoundary<2>::
+    boundary_velocity (const types::boundary_id ,
+                       const Point<2> &p) const
+    {
+      const LayeredFlowMaterial<2> *
+      material_model
+        = dynamic_cast<const LayeredFlowMaterial<2> *>(&this->get_material_model());
+      return AnalyticSolutions::LayeredFlow_velocity (p, material_model->get_beta(),
+                                                      material_model->get_epsilon());
+    }
+
+
+
+    template <>
+    Tensor<1,3>
+    LayeredFlowBoundary<3>::
+    boundary_velocity (const types::boundary_id ,
+                       const Point<3> &) const
+    {
+      Assert (false, ExcNotImplemented());
+      return Tensor<1,3>();
+    }
+
+
+
+
+    /**
+     *gravity model for the LayeredFlow benchmark
+    */
+
+    template <int dim>
+    class LayeredFlowGravity : public aspect::GravityModel::Interface<dim>
+    {
+      public:
+        virtual Tensor<1,dim> gravity_vector (const Point<dim> &pos) const;
+
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+        */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        double beta;
+        double epsilon;
+    };
+
+
+    /**
+      * A postprocessor that evaluates the accuracy of the solution.
+      *
+      * The implementation of error evaluators that correspond to the
+      * benchmarks defined in the paper Duretz et al. reference above.
+      */
+    template <int dim>
+    class LayeredFlowPostprocessor : public Postprocess::Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Generate graphical output from the current solution.
+         */
+        virtual
+        std::pair<std::string,std::string>
+        execute (TableHandler &statistics);
+
+      private:
+        /**
+         * Calculate the L2 dynamic topography error.
+         */
+        //double
+        //compute_dynamic_topography_error() const;
```

Comment:
just remove the whole private section here

Reply:
done